### DISCUSSION TITLE
Undo/Redo support for erase: Is it possible to programmatically erase a portion of fabric.js object ?

### DISCUSSION THREAD

**Discussion by VenkiNagaraj** [fabricjs/fabric.js#10291] [Q&A]
### Say, I have a rect fabric.js object. Is it possible to programmatically erase a portion of this object ? 

Basically, I'm trying to implement undo/redo feature in my app. When redo is clicked I have to programmatically erase a portion of the object. How can I achieve this ?
**Note: I want to store the undo/redo stacks in localStorage, so I have to stringify the stacks before adding to localStorage. Keep in mind some properties of fabric.js will get removed(eraser property) while stringifying.**

**Comment by tpvuser** [comment]
Did u check suggestion made here if it helps: https://github.com/fabricjs/fabric.js/issues/10011

**Comment by asturur** [comment]
Check the erase2d package from shaman123

**Comment by Rohan1348** [comment]
I have been implementing this feature recently and it appears to be similar as of yours.
(undo-redo for eraser - pop and store the eraser paths when doing undo and reuse these paths during the redo operation)
After storing the undo-redo stacks in local storage, for redo to work, we need to convert the string object (from local storage) back to fabric object.

```
(target = canvasObject)
target.eraser._objects.push(eraserPath);
target.eraser.set('dirty', true);
target.set('dirty', true);
```
                    
hope this helps.

**For conversion of a string object to fabric object refer to [https://jsfiddle.net/23has9df/]**
